perf/libunwind.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Pooja <pooja@linux.vnet.ibm.com>
+
+import os
+import re
+
+from avocado import Test
+from avocado import main
+from avocado.utils import archive, build, distro, process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Libunwind(Test):
+
+    def setUp(self):
+        '''
+        Build Libunwind library
+        Source:
+        https://github.com/pathscale/libunwind/archive/vanilla_pathscale.zip
+        '''
+        dist = distro.detect()
+        smm = SoftwareManager()
+        deps = ['gcc', 'libtool', 'autoconf', 'automake', 'make']
+        if dist.name == 'Ubuntu':
+            deps.extend(['dh-autoreconf', 'dh-dist-zilla', 'g++',
+                         'texlive-extra-utils'])
+        elif dist.name in ['SuSE', 'rhel', 'fedora', 'redhat']:
+            deps.extend(['gcc-c++'])
+        else:
+            self.fail('No packages list for your distro.')
```

Comment:
i think it can be `test not supported in distro ` as well please use `self.cancel `